s2/cellunion.go

```diff
@@ -569,7 +569,7 @@ func (cu *CellUnion) decode(d *decoder) {
 		d.err = fmt.Errorf("only version %d is supported", encodingVersion)
 		return
 	}
-	n := d.readInt64()
+	n := d.readUint64()
```

Comment:
Looks like this was the only use of `readInt64()`, so you can delete it.

https://github.com/search?q=repo%3Agolang%2Fgeo%20readInt64&type=code